apps/admin/frontend/src/components/reporting/group_by_editor.tsx

```diff
@@ -0,0 +1,84 @@
+import { Tabulation } from '@votingworks/types';
+import styled from 'styled-components';
+import { Checkbox, Font } from '@votingworks/ui';
+
+type Grouping = keyof Tabulation.GroupBy;
+
+function getAllowedGroupings(): Grouping[] {
+  return ['groupByPrecinct', 'groupByVotingMethod', 'groupByBallotStyle'];
+}
+
+const GROUPING_LABEL: Record<Grouping, string> = {
+  groupByParty: 'Party',
+  groupByPrecinct: 'Precinct',
+  groupByBallotStyle: 'Ballot Style',
+  groupByVotingMethod: 'Voting Method',
+  groupByBatch: 'Batch',
+  groupByScanner: 'Scanner',
+};
+
+export interface GroupByEditorProps {
+  groupBy: Tabulation.GroupBy;
+  setGroupBy: (groupBy: Tabulation.GroupBy) => void;
+}
+
+const Container = styled.div`
+  display: grid;
+  grid-template-columns: repeat(4, min-content);
+  gap: 1rem;
+`;
+
+const Item = styled.span`
+  margin-bottom: 0.5rem;
+  display: flex;
+  flex-wrap: nowrap;
+  align-items: center;
+  gap: 0.5rem;
+`;
+
+const ItemLabel = styled(Font)`
+  white-space: nowrap;
+`;
+
+const CheckboxContainer = styled.button`
+  border: none;
+  background: none;
+  padding: 0;
+  margin: 0;
+  flex-grow: 0;
+  flex-shrink: 0;
+  font-size: 0.8em;
+`;
+
+export function GroupByEditor({
+  groupBy,
+  setGroupBy,
+}: GroupByEditorProps): JSX.Element {
+  function toggleGrouping(grouping: Grouping): void {
+    setGroupBy({
+      ...groupBy,
+      [grouping]: !groupBy[grouping],
+    });
+  }
+
+  const allowedGroupings = getAllowedGroupings();
+  return (
+    <Container data-testid="group-by-editor">
+      {allowedGroupings.map((grouping) => {
+        const checked = groupBy[grouping];
+        return (
+          <Item key={grouping}>
+            <CheckboxContainer
+              onClick={() => toggleGrouping(grouping)}
+              aria-label={`Report By ${GROUPING_LABEL[grouping]}`}
+              aria-pressed={checked}
+            >
+              <Checkbox checked={checked} />
+            </CheckboxContainer>
+            <ItemLabel>{GROUPING_LABEL[grouping]}</ItemLabel>
```

Comment:
Does the label then need some kind of outline or other indicator of the clickable box, or is it OK to just simply make it clickable?

Reply:
Ah, good question. Technically, yes, but I don't think that's crucial - a lot of implementations I've seen don't include any overt hover state beyond the cursor changing to `pointer` on hover (which I think we automatically get here since `CheckboxContainer` is a `button`).

If you're curious, here's what I used as reference while working on the contest "checkboxes": https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/examples/checkbox/

Reply:
OK, make the label clickable [e4e2b27](https://github.com/votingworks/vxsuite/pull/3927/commits/e4e2b27ad1ff69fd374d26197ee8af0d1e422c9c)!

I saw that the article lists a number of other semantic changes which I haven't incorporated yet.

Reply:
It would be nice to incorporate other hover / focus markers, but often those are dotted lines, which we use around disabled buttons, so I've been hesitant to re-use them for focus.

Reply:
Yeah, +1 - we can leave the other stuff out for now, IMO. Might be worth building out a shared checkbox list component in the future, then we can address the a11y nits there. There's an open task around adding a more flexible theme that will make that easier too, so we can block on that